File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Bomb.java
package com.nopalsoft.ponyrace.game_objects;

import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.physics.box2d.Body;
import com.esotericsoftware.spine.Skeleton;
import com.nopalsoft.ponyrace.Settings;
import com.nopalsoft.ponyrace.game.TileMapHandler;

public class Bomb extends BaseGameObject {
    public static final float TIEMPO_NORMAL = 1.5f;
    public static final float TIEMPO_EXPLOSION = .3f;
    public final float TIEMPO_HURT;
    public float lastStatetime;
    public float stateTime;
    public float angulo;
    public State state;
    public Skeleton skelBomb;

    public Bomb(float x, float y, TileMapHandler world) {
        super(x, y);
        stateTime = 0;
        lastStatetime = stateTime;
        state = State.normal;
        skelBomb = new Skeleton(world.game.assetsHandler.skeletonBombData);
        skelBomb.setToSetupPose();

        switch (Settings.bombLevel) {

            case 1:
                TIEMPO_HURT = 2.5f;
                break;
            case 2:
                TIEMPO_HURT = 2.7f;
                break;
            case 3:
                TIEMPO_HURT = 3f;
                break;
            case 4:
                TIEMPO_HURT = 3.25f;
                break;
            case 5:
                TIEMPO_HURT = 3.5f;
                break;
            case 0:
            default:
                TIEMPO_HURT = 2;
                break;
        }
    }

    public void update(float delta, Body obj) {
        lastStatetime = stateTime;
        stateTime += delta;

        position.x = obj.getPosition().x;
        position.y = obj.getPosition().y;
        angulo = MathUtils.radiansToDegrees * obj.getAngle();

        if (state == State.normal && stateTime >= TIEMPO_NORMAL) {
            state = State.explode;
            stateTime = 0;
        }
    }

    public void explode(Body obj) {
        if (state == State.normal) {
            state = State.explode;
            stateTime = 0;
            obj.setLinearVelocity(0, 0);
        }
    }

    public enum State {
        normal,
        explode
    }
}

File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Bomb.kt
package com.nopalsoft.ponyrace.game_objects

import com.badlogic.gdx.math.MathUtils
import com.badlogic.gdx.physics.box2d.Body
import com.esotericsoftware.spine.Skeleton
import com.nopalsoft.ponyrace.Settings
import com.nopalsoft.ponyrace.game.TileMapHandler

class Bomb(x: Float, y: Float, world: TileMapHandler) : BaseGameObject(x, y) {
    @JvmField
    val bombTimerSeconds = when (Settings.bombLevel) {
        1 -> 2.5F
        2 -> 2.7F
        3 -> 3F
        4 -> 3.25F
        5 -> 3.5F
        0 -> 2F
        else -> 2F
    }

    @JvmField
    var stateTime = 0F

    @JvmField
    var lastStatetime = stateTime

    var angulo = 0F

    @JvmField
    var state = State.NORMAL

    @JvmField
    var skelBomb = Skeleton(world.game.assetsHandler!!.skeletonBombData)

    init {
        skelBomb.setToSetupPose()
    }

    fun update(delta: Float, obj: Body) {
        lastStatetime = stateTime
        stateTime += delta

        position.x = obj.getPosition().x
        position.y = obj.getPosition().y
        angulo = MathUtils.radiansToDegrees * obj.angle

        if (state == State.NORMAL && stateTime >= TIEMPO_NORMAL) {
            state = State.EXPLODE
            stateTime = 0f
        }
    }

    fun explode(obj: Body) {
        if (state == State.NORMAL) {
            state = State.EXPLODE
            stateTime = 0f
            obj.setLinearVelocity(0f, 0f)
        }
    }

    enum class State {
        NORMAL,
        EXPLODE
    }

    companion object {
        const val TIEMPO_NORMAL = 1.5F
        const val TIEMPO_EXPLOSION = .3F
    }
}
